langchain/vectorstores/vectara.py

```diff
@@ -0,0 +1,328 @@
+"""Wrapper around Vectara vector database."""
+from __future__ import annotations
+
+import json
+import logging
+from hashlib import md5
+from typing import Any, Iterable, List, Optional, Tuple, Type
+
+import requests
+
+from langchain.embeddings.base import Embeddings
+from langchain.schema import BaseRetriever, Document
+from langchain.vectorstores.base import VectorStore
+
+
+class Vectara(VectorStore):
+    """Implementation of Vector Store using Vectara (https://vectara.com)
+    Example:
+        .. code-block:: python
+
+            from langchain.vectorstores import Vectara
+            vectorstore = Vectara(customer_id, corpus_id, api_key)
+    """
+
+    def __init__(
+        self, vectara_customer_id: int, vectara_corpus_id: int, vectara_api_key: str
+    ):
+        """Initialize with Vectara API."""
+        self._vectara_customer_id = vectara_customer_id
+        self._vectara_corpus_id = vectara_corpus_id
+        self._vectara_api_key = vectara_api_key
```

Comment:
we usually make it possible to set credentials (like api key and other id's) via environment variables. if we wanted to support that here could d something like
```python
def __init__(vectara_api_key: Optional[str]=None, ...):
  self.vectara_api_key = vectara_api_key or get_from_env("vectara_api_key", "VECTARA_API_KEY")
  ...
```

Reply:
Good suggestion - updated!